UI/CombatReport/CombatReportWnd.cpp

```diff
@@ -10,12 +11,34 @@
 #include "CombatLogWnd.h"
 
 #include <GG/TabWnd.h>
+#include <GG/Layout.h>
 
 namespace {
     GG::X COMBAT_LOG_WIDTH(400);
     GG::Y COMBAT_LOG_HEIGHT(300);
-    GG::X RESIZE_MARGIN_X(4);
-    GG::Y RESIZE_MARGIN_Y(4);
+
+    // Duplicated from MapWnd.cpp - used here to find maximum size of the
+    // combat report window.
+    GG::X AppWidth() {
+        if(HumanClientApp* app = HumanClientApp::GetApp())
+            return app->AppWidth();
+        return GG::X0;
+    }
+
+    GG::Y AppHeight() {
+        if(HumanClientApp* app = HumanClientApp::GetApp())
+            return app->AppHeight();
+        return GG::Y0;
+    }
+
+    GG::Pt LimitToAppSize(const GG::Pt& sz_in) {
```

Comment:
It doesn't have a parent Wnd, as far as I can see.  I looked at putting the functionality into a class for other windows to use if needed (CUIWnd because it's FO-specific?) but I think that would fit better in another PR.  For now, this window just needed some way of getting the app's dimensions to make sure it didn't try to become bigger than the screen and I used that code from MapWnd.cpp as an example.  I've trimmed out the first two functions for the next commit, at least.